doc/examples/example5.md

```diff
@@ -0,0 +1,397 @@
+\code{.cpp}
+class EncryptionCtx {
+    ...
+    virtual std::vector<uint8_t> encrypt(const std::vector<uint8_t>& message) = 0;
+};
+
+class DecryptionCtx {
+    ...
+    virtual std::vector<uint8_t> decrypt(const std::vector<uint8_t>& message) = 0;
+}
+\endcode
+That is, these classes can be used to access a en-/decryption context in a generic manner.
+
+## Signature Interface
+
+All contexts that support signing implement a method to sign pre-hashed message digests (`signDigest()`) or unhashed messages (`signMessage()`) or both. Please note that if a message digest shall be signed (or verified) it is expected that the client provides the digest of the message to be signed (i.e. performs the hashing on his own). On the contrary, signing a message means that the message digest will be calculated and signed then. The classes `mococrw::DigestSignatureCtx`, `mococrw::MessageSignatureCtx`, `mococrw::DigestVerificationCtx` and `mococrw::MessageVerificationCtx` provide a pure virtual interface to these methods:
+
+\code{.cpp}
+class DigestSignatureCtx {
+    ...
+    virtual std::vector<uint8_t> signDigest(const std::vector<uint8_t> &messageDigest) = 0;
+};
+
+class MessageSignatureCtx {
+    ...
+    virtual std::vector<uint8_t> signMessage(const std::vector<uint8_t> &message) = 0;
+};
+
+class DigestVerificationCtx {
+    ...
+    virtual void verifyDigest(const std::vector<uint8_t> &signature,
+                              const std::vector<uint8_t> &digest) = 0;
+};
+
+class MessageVerificationCtx {
+    ...
+    virtual void verifyMessage(const std::vector<uint8_t> &signature,
+                               const std::vector<uint8_t> &message) = 0;
+};
+\endcode
+
+#### Example
+If a context supports both, signing messages and message digest, the following code snippets are equal. For simplicity it is assumes that SHA256 is used to create the signed hash.
+\code{.cpp}
+    std::shared_ptr<DigestSignatureCtx> ctx = ...; // Hash Function set to SHA256
+    std::vector<uint8_t> message = {'H', 'e', 'l', 'l', 'o', ' ', 'W', 'o', 'r', 'l', 'd'};
+    std::vector<uint8_t> digest = mococrw::Hash::sha256(message);
+    std::vector<uint8_t> signature = ctx->signDigest(digest);
+\endcode
+\code{.cpp}
+    std::shared_ptr<MessageSignatureCtx> = ...; // Hash Function set to SHA256
+    std::vector<uint8_t> message = {'H', 'e', 'l', 'l', 'o', ' ', 'W', 'o', 'r', 'l', 'd'};
+    std::vector<uint8_t> siganture = ctx->signMessage(message)
+\endcode
+
+The verification contexts behave similarly.
+
+# RSA - Encryption
+
+By default the RSA encryption crypto contexts (for en- and decryption) are using the following options:
+ * OAEP Padding:
+    * Hash Function SHA256
+    * Mask Generation Function: MGF1(SHA256)
+    * Empty label
+
+## Encryption (Default Padding)
+
+\code{.cpp}
+std::string pubKey= R"(-----BEGIN PUBLIC KEY-----
+...
+-----END PUBLIC KEY-----)";
+
+std::vector<uint8_t> encryptedMessage = {...};
```

Comment:
This should have been `message`, right? You're declaring `std::vector<uint8_t> encryptedMessage` below with the result of the encryption operation.